core/state/statedb.go

```diff
@@ -1556,8 +1611,12 @@ func (s *StateDB) SnapToDiffLayer() ([]common.Address, []types.DiffAccount, []ty
 	for accountHash, storage := range s.snapStorage {
 		keys := make([]string, 0, len(storage))
 		values := make([][]byte, 0, len(storage))
-		for k, v := range storage {
+		for k := range storage {
 			keys = append(keys, k)
+		}
+		sort.Strings(keys)
```

Comment:
why sort is needed here?

Reply:
> 



> why sort is needed here?

we should sort here to ensure the keys is sorted for each account.

Reply:
but why keys have to be sorted?

Reply:
> but why keys have to be sorted? This is on the critical path which impacts the performance, need a strong reason here.

@NashBC

Reply:
Have moved into cacheDiffLayer()